renderdoc/driver/shaders/dxbc/dxbc_debug.cpp

```diff
@@ -2088,175 +2083,26 @@ State State::GetNext(GlobalState &global, State quad[4]) const
-        return s;
-      }
-
-      context->CSSetConstantBuffers(0, 1, &constBuf);
-
-      context->CSSetShader(cs, NULL, 0);
-
-      ID3D11UnorderedAccessView *uav = NULL;
-
-      ID3D11Buffer *uavBuf = NULL;
-      ID3D11Buffer *copyBuf = NULL;
-
-      D3D11_BUFFER_DESC bdesc;
-
-      bdesc.BindFlags = D3D11_BIND_UNORDERED_ACCESS;
-      bdesc.CPUAccessFlags = 0;
-      bdesc.MiscFlags = 0;
-      bdesc.StructureByteStride = sizeof(Vec4f);
-      bdesc.ByteWidth = sizeof(Vec4f) * 2;
-      bdesc.Usage = D3D11_USAGE_DEFAULT;
-
-      hr = device->CreateBuffer(&bdesc, NULL, &uavBuf);
-
-      if(FAILED(hr))
-      {
-        RDCERR("Failed to create UAV buf HRESULT: %s", ToStr(hr).c_str());
-        return s;
-      }
-
-      bdesc.BindFlags = 0;
-      bdesc.CPUAccessFlags = D3D11_CPU_ACCESS_READ;
-      bdesc.Usage = D3D11_USAGE_STAGING;
-
-      hr = device->CreateBuffer(&bdesc, NULL, &copyBuf);
-
-      if(FAILED(hr))
-      {
-        RDCERR("Failed to create copy buf HRESULT: %s", ToStr(hr).c_str());
-        return s;
-      }
-
-      D3D11_UNORDERED_ACCESS_VIEW_DESC uavDesc = {};
-
-      uavDesc.Format = DXGI_FORMAT_R32G32B32A32_FLOAT;
-      uavDesc.ViewDimension = D3D11_UAV_DIMENSION_BUFFER;
-      uavDesc.Buffer.FirstElement = 0;
-      uavDesc.Buffer.NumElements = 2;
-      uavDesc.Buffer.Flags = 0;
-
-      hr = device->CreateUnorderedAccessView(uavBuf, &uavDesc, &uav);
-
-      if(FAILED(hr))
-      {
-        RDCERR("Failed to create uav HRESULT: %s", ToStr(hr).c_str());
-        return s;
-      }
-
-      context->CSSetUnorderedAccessViews(0, 1, &uav, NULL);
-      context->Dispatch(1, 1, 1);
-
-      context->CopyResource(copyBuf, uavBuf);
-
-      D3D11_MAPPED_SUBRESOURCE mapped;
-      hr = context->Map(copyBuf, 0, D3D11_MAP_READ, 0, &mapped);
-
-      if(FAILED(hr))
-      {
-        RDCERR("Failed to map results HRESULT: %s", ToStr(hr).c_str());
-        return s;
-      }
-
       ShaderVariable calcResultA("calcA", 0.0f, 0.0f, 0.0f, 0.0f);
       ShaderVariable calcResultB("calcB", 0.0f, 0.0f, 0.0f, 0.0f);
-
-      uint32_t *resA = (uint32_t *)mapped.pData;
-      uint32_t *resB = resA + 4;
-
-      memcpy(calcResultA.value.uv, resA, sizeof(uint32_t) * 4);
-      memcpy(calcResultB.value.uv, resB, sizeof(uint32_t) * 4);
-
-      context->Unmap(copyBuf, 0);
-
-      SAFE_RELEASE(constBuf);
-      SAFE_RELEASE(uavBuf);
-      SAFE_RELEASE(copyBuf);
-      SAFE_RELEASE(uav);
-      SAFE_RELEASE(cs);
-
-      // restore whatever was on CS slot 0 before we messed with it
-
-      UINT append[] = {~0U};
-      context->CSSetConstantBuffers(0, 1, &prevCB);
-      context->CSSetUnorderedAccessViews(0, 1, &prevUAV, append);
-
-      SAFE_RELEASE(context);
-
-      SAFE_RELEASE(prevCB);
-      SAFE_RELEASE(prevUAV);
-
-      if(op.operation == OPCODE_SINCOS)
+      if(apiWrapper->CalculateMathIntrinsic(op.operation, srcOpers[0], calcResultA, calcResultB))
```

Comment:
This will need to pass `srcOpers[1]` for `OPCODE_SINCOS` I think? Because `srcOpers` are fetched from `operands[1]` onwards without considering that this operation has two outputs so `operands[1]` is the second output.

Reply:
Great catch, fixed!